rust/sdk-examples/src/utils/database.rs

```diff
@@ -179,16 +179,19 @@ where
     debug!("Executing query: {:?}", debug_string);
     let conn = &mut pool.get().await.map_err(|e| {
         warn!("Error getting connection from pool: {:?}", e);
-        diesel::result::Error::DatabaseError(
-            diesel::result::DatabaseErrorKind::UnableToSendCommand,
-            Box::new(e.to_string()),
-        )
+        ProcessorError::DBStoreError {
```

Comment:
is it possible to move debug string construction only when erroring out? and is it possible to remove the clone?

Reply:
I think this only runs on error, this is a closure.

Reply:
oh my question is, can we move this into closure?  `let debug_string = diesel::debug_query::<Backend, _>(&final_query).to_string();`  my concern is the to_string creates a new copy of data, which is not only under error case.

Reply:
I just took out the debug logs and now it only prints on error 👍

Reply:
love this